geode-for-redis/src/main/java/org/apache/geode/redis/internal/statistics/GeodeRedisStats.java

```diff
@@ -218,12 +248,21 @@ private static void fillListWithCompletedCommandDescriptors(StatisticsTypeFactor
     }
   }
 
-  private static void fillCompletedIdMap() {
-    for (RedisCommandType command : RedisCommandType.values()) {
+  private static void fillCompletedIdMap(RedisCommandType.Category category) {
+    for (RedisCommandType command : RedisCommandType.getCommandsForCategory(category)) {
       String name = command.name().toLowerCase();
       String statName = name + "Completed";
+      completedCommandStatIds.put(command,
+          statisticTypes.get(command.category().name()).nameToId(statName));
```

Comment:
I've moved the lookup outside the loop, however I'm still passing in `category` since that is used to do the lookup on which commands are in the category.

Reply:
Now passing in both category and type.